To: vim_dev@googlegroups.com
Subject: Patch 7.3.186
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.186
Problem:    When 'clipboard' contains "unnamed" or "unnamedplus" the value of
	    v:register is wrong for operators without a specific register.
Solution:   Adjust the register according to 'clipboard'. (Ingo Karkat)
Files:	    src/normal.c


*** ../vim-7.3.185/src/normal.c	2010-12-17 18:52:56.000000000 +0100
--- src/normal.c	2011-05-10 16:07:49.000000000 +0200
***************
*** 1202,1208 ****
      {
  	clearop(oap);
  #ifdef FEAT_EVAL
! 	set_reg_var('"');
  #endif
      }
  
--- 1202,1214 ----
      {
  	clearop(oap);
  #ifdef FEAT_EVAL
! 	{
! 	    int regname = 0;
! 	    /* Adjust the register according to 'clipboard', so that when
! 	     * "unnamed" is present it becomes '*' or '+' instead of '"'. */
! 	    adjust_clip_reg(&regname);
! 	    set_reg_var(regname);
! 	}
  #endif
      }
  
*** ../vim-7.3.185/src/version.c	2011-05-10 16:00:43.000000000 +0200
--- src/version.c	2011-05-10 16:10:10.000000000 +0200
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     186,
  /**/

-- 
hundred-and-one symptoms of being an internet addict:
80. At parties, you introduce your spouse as your "service provider."

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
